packages/helix_editor.rb

```diff
@@ -0,0 +1,73 @@
+require 'package'
+
+class Helix_editor < Package                 # The first character of the class name must be upper case
+  description 'A vim like editor with LSP support'
+  homepage 'https://helix-editor.com/'
+  version '22.12'
+  license 'MPL-2.0' # license of source
+  compatibility 'all'
+  source_url 'https://github.com/helix-editor/helix/archive/refs/tags/22.12.tar.gz'
+  source_sha256 'edae8af46401b45c3e71c38b4fa99f931c4458127978ccd1b29aaae79331d972'   # Use the command "sha256sum"
+  #source_url 'file:///usr/local/home/helix/deleteme.tar.gz'
+  #source_sha256 'c3a0e022a7bcd6b139ae7abf1e0dae33c8f9ec82dc680076d9e5ea6593b85bc6'
+
+  depends_on 'rust' => :build
+
+  # TODO is it the appropriate location?
+  # By default, helix expects runtime to be in ~/.config or XDG_CONFIG_HOME.
+  # The first location is not appropriate because helix will not be installed in ~/
+  # With the second location crew complaints that it is not  FHS3 compliant.
+  @runtime_dest_dir_suffix = "/var/lib/helix"
+
+  def self.build
+   puts "Building. This may be long."
+   system "cargo build \
+      --release \
+      --locked \
+     "
+  end
+
+  def self.install
+    # Copy executable
+    helix_executable_dest_dir = "#{CREW_DEST_PREFIX}/bin"
+    FileUtils.mkdir_p("#{helix_executable_dest_dir}")
+    FileUtils.cp("./target/release/hx", "#{helix_executable_dest_dir}")
+    # Copy runtime dir
+    FileUtils.mkdir_p("#{CREW_DEST_PREFIX}#{@runtime_dest_dir_suffix}")
+    FileUtils.cp_r("./runtime", "#{CREW_DEST_PREFIX}#{@runtime_dest_dir_suffix}")
+    # Set env var for runtime dir
+    env_dir = "#{CREW_DEST_PREFIX}/etc/env.d"
+    FileUtils.mkdir_p "#{env_dir}"
+    File.write("#{env_dir}/helix",
+               "export HELIX_RUNTIME=#{CREW_PREFIX}#{@runtime_dest_dir_suffix}/runtime")
+    # system "source #{env_dir}/helix"
+  end
+
+  def self.check
+    # # TODO (verify that this check works, and if it fails when the system language
+    # #  is not english)
+    # helix_health = capture(:stdout) { system("hx --health") }
+    # return !helix_health.contains('Runtime directory does not exist')
+    return true
```

Comment:
A simple test could be to run the executable to see if it provides any output.  Something like `helix --version`.

Reply:
I tried 'hx --version'. However it seems the self.check method is never called. I don't really understand why.

Reply:
I tried 'hx --version'. However it seems the self.check method is never called. I don't really understand why.

Reply:
self.check is only used during `crew build`, not `crew install -s`...

Reply:
Sorry @ncdm-stldr:
I probably should have explained better how to do it.  Something like below:
```
def self.check
  hx_ver = `hx --version 2>&1`
  return (hx_ver.empty? || hx_ver.include?('No such file or directory') || hx_ver.include?('not found')) ? false : true
end
```

Reply:
Ok, I tried a `crew build` and saw it was indeed called there. Thanks.

I tried something along the lines of the suggestion. However even if  self.check returns false, the build is not aborted. I think that I will remove the self.check method; it is not very important.

Reply:
Try explicitly throwing an error if the test fails.

Reply:
Ok. Yes it works with the raise error.
(I tried this once but I forgot to copy the helix_editor.rb at the correct location...)